#! /bin/sh -e
## 20_ps.1.dpatch by Craig Small <csmall@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix ps manual page

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argum
ent"
                exit 1;;
esac

exit 0
@DPATCH@
--- procps/ps/ps.1.orig	Thu Apr  1 22:20:02 2004
+++ procps/ps/ps.1	Thu Apr  1 22:23:09 2004
@@ -1,3 +1,5 @@
+'\" t
+.\" (The preceding line is a note to broken versions of man to tell
 .\" Man page for ps.
 .\" Quick hack conversion by Albert Cahalan, 1998.
 .\" Licensed under version 2 of the Gnu General Public License.
@@ -10,8 +12,8 @@
 .\" invented this crap in 1973. Oh yeah, they did. Sorry.
 .\"
 .TH PS 1 "July 5, 1998" "Linux" "Linux User's Manual"
-.SH \fRNAME\fR
-ps \- report process status
+.SH NAME
+ps \- report a snapshot of the current processes.
 .ad r
 .na
 .ss 12 0
@@ -19,17 +21,17 @@
 .nh
 .nf
 
-SYNOPSIS
+.SH SYNOPSIS
 ps [options]
 
 
-DESCRIPTION
+.SH DESCRIPTION
 ps gives a snapshot of the current processes. If you want
 a repetitive update of this status, use top. This man
 page documents the /proc-based version of ps, or tries to.
 
 
-COMMAND-LINE OPTIONS
+.SH "COMMAND-LINE OPTIONS"
 
 This version of ps accepts several kinds of options.
 
@@ -43,7 +45,9 @@
 when options are preceeded by a dash. The PS_PERSONALITY environment
 variable (described below) provides more detailed control of ps behavior.
 
-SIMPLE PROCESS SELECTION
+.SH "SIMPLE PROCESS SELECTION"
+.TS
+l l.
 -A           select all processes
 -N           negate selection
 -a           select all with a tty except session leaders
@@ -55,8 +59,11 @@
 r            restrict output to running processes
 x            select processes without controlling ttys
 --deselect   negate selection
+.TE
 
-PROCESS SELECTION BY LIST
+.SH "PROCESS SELECTION BY LIST"
+.TS
+l l.
 -C           select by command name
 -G           select by RGID (supports names)
 -U           select by RUID (supports names)
@@ -78,8 +85,10 @@
 --user       select by effective user name or ID
 -123         implied --sid
 123          implied --pid
-
-OUTPUT FORMAT CONTROL
+.TE
+.SH "OUTPUT FORMAT CONTROL"
+.TS
+l l.
 -O           is preloaded "-o"
 -F           extra full format
 -M           add column for security data
@@ -100,8 +109,10 @@
 v            display virtual memory format
 --format     user-defined format
 --context    display security context format (NSA SELinux, etc.)
-
-OUTPUT MODIFIERS
+.TE
+.SH "OUTPUT MODIFIERS"
+.TS
+l l.
 -H           show process hierarchy (forest)
 -n           set namelist file
 -w           wide output
@@ -125,31 +136,40 @@
 --rows       set screen height
 --sort       specify sorting order
 --width      set screen width
-
-THREAD DISPLAY
+.TE
+.SH "THREAD DISPLAY"
+.TS
+l l.
 -L           show threads, possibly with LWP and NLWP columns
 -T           show threads, possibly with SPID column
 -m           show threads after processes
 H            show threads as if they were processes
 m            show threads after processes
+.TE
 
-INFORMATION
+.SH INFORMATION
+
+.TS
+l l.
 -V          print version
 L           list all format specifiers
 V           show version info
 --help      print help message
 --info      print debugging info
 --version   print version
+.TE
+
+.SH OBSOLETE
 
-OBSOLETE
+.TS
+l l.
 A           increases the argument space (DecUnix)
 M           use alternate core (try -n or N instead)
 W           get swap info from ... not /dev/drum (try -n or N instead)
 k           use /vmcore as c-dumpfile (try -n or N instead)
-
-
-
-NOTES
+.TE
+  
+.SH NOTES
 
 User-defined format options ("o", "-o", "O", and "-O") offer
 a way to specify individual output columns. Headers may be
@@ -251,14 +271,17 @@
 will be destroyed by init(8) if the parent process exits.
 
 
-PROCESS FLAGS
-
+.SH "PROCESS FLAGS"
+.TS
+l l l.
 FORKNOEXEC   1    forked but didn't exec
 SUPERPRIV    4    used super-user privileges
+.TE
 
+.SH "PROCESS STATE CODES"
 
-PROCESS STATE CODES
-
+.TS
+l l.
 D uninterruptible sleep (usually IO)
 R runnable (on run queue)
 S sleeping
@@ -266,23 +289,31 @@
 W paging (2.4 kernels and older only)
 X dead
 Z a defunct ("zombie") process
+.TE
 
 For BSD formats and when the "stat" keyword is used, additional
 characters may be displayed:
 
-< high-priority task
-N low-priority task
-L has pages locked into memory (for real-time and custom IO)
-s is a session leader
-+ is in the foreground process group
+.IP <
+high-priority task
+.IP N
+low-priority task
+.IP L
+has pages locked into memory (for real-time and custom IO)
+.IP s
+is a session leader
+.IP +
+is in the foreground process group
 
-SORT KEYS
+.SH "SORT KEYS"
 
 Note that the values used in sorting are the internal values ps uses and not
 the `cooked' values used in some of the output format fields. Pipe ps
 output into the sort(1) command if you want to sort the cooked values.
 
-KEY LONG       DESCRIPTION
+.TS
+l l l.
+\fBKEY LONG       DESCRIPTION\fR
 c   cmd        simple name of executable
 C   cmdline    full command line
 f   flags      flags as in long format F field
@@ -309,14 +340,15 @@
 u   user       user name
 v   vsize      total VM size in kB
 y   priority   kernel scheduling priority
+.TE
 
-
-AIX FORMAT DESCRIPTORS
+.SH "AIX FORMAT DESCRIPTORS"
 
 This ps supports AIX format descriptors, which work somewhat like the
 formatting codes of printf(1) and printf(3). For example, the normal
 default output can be produced with this:   ps -eo "%p %y %x %c"
-
+.TS
+l l l.
 CODE  NORMAL    HEADER
 %C    pcpu      %CPU
 %G    group     GROUP
@@ -333,14 +365,16 @@
 %x    time      TIME
 %y    tty       TTY
 %z    vsz       VSZ
+.TE
 
-
-STANDARD FORMAT SPECIFIERS
+.SH "STANDARD FORMAT SPECIFIERS"
 
 These may be used to control both output format and sorting.
 For example:  ps -eo pid,user,args --sort user
 
-CODE         HEADER
+.TS
+l l.
+\fBCODE         HEADER\fR
 %cpu         %CPU    
 %mem         %MEM    
 alarm        ALARM   
@@ -461,25 +495,51 @@
 vsize        VSZ     
 vsz          VSZ     
 wchan        WCHAN   
+.TE
 
+.SH "ENVIRONMENT VARIABLES"
 
-
-
-ENVIRONMENT VARIABLES
 The following environment variables could affect ps:
-    COLUMNS             Override default display width.
-    LINES               Override default display height.
-    PS_PERSONALITY      Set to one of posix,old,linux,bsd,sun,digital...
-    CMD_ENV             Set to one of posix,old,linux,bsd,sun,digital...
-    I_WANT_A_BROKEN_PS  Force obsolete command line interpretation.
-    LC_TIME             Date format.
-    PS_COLORS           Not currently supported.
-    PS_FORMAT           Default output format override.
-    PS_SYSMAP           Default namelist (System.map) location.
-    PS_SYSTEM_MAP       Default namelist (System.map) location.
-    POSIXLY_CORRECT     Don't find excuses to ignore bad "features".
-    UNIX95              Don't find excuses to ignore bad "features".
-    _XPG                Cancel CMD_ENV=irix non-standard behavior.
+
+.TP 
+.B COLUMNS
+Override default display width.
+.TP
+.B LINES
+Override default display height.
+.TP
+.B PS_PERSONALITY
+Set to one of posix,old,linux,bsd,sun,digital...
+.TP
+.B CMD_ENV
+Set to one of posix,old,linux,bsd,sun,digital...
+.TP
+.B I_WANT_A_BROKEN_PS
+Force obsolete command line interpretation.
+.TP
+.B LC_TIME
+Date format.
+.TP
+.B PS_COLORS
+Not currently supported.
+.TP
+.B PS_FORMAT
+Default output format override.
+.TP
+.B PS_SYSMAP
+Default namelist (System.map) location.
+.TP
+.B PS_SYSTEM_MAP
+Default namelist (System.map) location.
+.TP
+.B POSIXLY_CORRECT
+Don't find excuses to ignore bad "features".
+.TP
+.B UNIX95
+Don't find excuses to ignore bad "features".
+.TP
+.B _XPG
+Cancel CMD_ENV=irix non-standard behavior.
 
 In general, it is a bad idea to set these variables. The one exception
 is CMD_ENV or PS_PERSONALITY, which could be set to Linux for normal
@@ -487,7 +547,9 @@
 of the Unix98 standard.
 
 
-PERSONALITY
+.SH "PERSONALITY"
+.TS
+l l.
     390      like the S/390 OpenEdition ps
     aix      like AIX ps
     bsd      like FreeBSD ps (totally non-standard)
@@ -509,9 +571,9 @@
     unix     standard
     unix95   standard
     unix98   standard
+.TE
 
-
-EXAMPLES
+.SH "EXAMPLES"
 To see every process on the system using standard syntax:
     ps -e
 To see every process on the system using BSD syntax:
@@ -525,13 +587,13 @@
 Print only the process IDs of syslogd:
     ps -C syslogd -o pid=
 
-SEE ALSO
-top(1) pgrep(1) pstree(1) proc(5)
+.SH "SEE ALSO"
+.BR top (1), pgrep (1), pstree (1),  proc (5).
 
-STANDARDS
+.SH STANDARDS
 This ps conforms to version 2 of the Single Unix Specification.
 
-AUTHOR
+.SH AUTHOR
 ps was originally written by Branko Lankester <lankeste@fwi.uva.nl>. Michael
 K. Johnson <johnsonm@redhat.com> re-wrote it significantly to use the proc
 filesystem, changing a few things in the process. Michael Shields
@@ -543,4 +605,4 @@
 <acahalan@cs.uml.edu> rewrote ps for full Unix98 and BSD support, along with
 some ugly hacks for obsolete and foreign syntax.
 
-Please send bug reports to <acahalan@cs.uml.edu>
+Please send bug reports to <procps-feedback@lists.sf.net>.
